pkg/opa/opa.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 type Client interface {
-	QueryPermissions(resource string, action Action, ids []string) (bool, error)
+	QueryPermissions(resource string, action Action, ids []string, overrideHeaderValue string) (bool, error)
```

Comment:
Id say change the interface to receive `queryPermissionOptions` and have those fields on that structure.

everytime you add a field, you break the interface and need to add on each implementor, lets cut that pattern and make it a struct